### PR TITLE
Milestone 6: Views

### PR DESCRIPTION
**Pull request by DuaneDave**
### In this PR, I added:

- Logic to the post index and show
- Logic to the users index and show
- Partials for the users and view template

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,8 @@
-<h1> Hello views </h1>
\ No newline at end of file
+<h1> List all users </h1> 
+<ul class='list'>
+<% @users.each do |user| %>
+<a href='/users/<%= user.id %>'>
+   <%= render 'partials/user', user: user %>
+<a>
+<% end %>
+</ul>
```
Comment (tresorsawasawa) [inline app/views/users/index.html.erb]: - [ ] As required for the project, kindly make sure your this index view page design sticks as much as possible to this **[wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_root_users.png?token=GHSAT0AAAAAAB3P7JH4IOK7UJHG5ZSIIAREY4HXWQA)**. Right now there is no styling to this page

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,15 @@
...
+<ul class='card'>
+    <% @user.recent_three_posts.each do |post| %>
+      <a href="/users/<%= @user.id %>/posts/<%= post.id %> ">
+        <%= render 'partials/post', post: post %>
+      </a>
+    <% end %>
+</ul>
+
+  <%= link_to "See all posts", user_posts_path(@user.id)%>
+</div>
```
Comment (tresorsawasawa) [inline app/views/users/show.html.erb]: - [ ] Kindly make sure your `app/views/users/show.html.erb` page design sticks as much as possible to this **[wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_page.png?token=GHSAT0AAAAAAB3P7JH5ZHVOSISGQEF7TBNKY4HYGAA)** for the project's requirements. Right now there is no styling provided yet to this page

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,11 @@
...
+  <%= render'partials/user', user: @user %>
+    <% @posts.each do |post| %>
+   <a href="/users/<%= @user.id %>/posts/<%= post.id %>">
+    <ul class="post-list">
+    <%= render 'partials/post', post: post %></ul>
+    </a>
+    </ul>
+    <% end %>
+  <button type='button'>Pagination</button>
+</section>
```
Comment (tresorsawasawa) [inline app/views/posts/index.html.erb]: - [ ] As required for the project, kindly make sure the `app/views/posts/index.html.erb` page design sticks as much as possible to this **[wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png)**. Right now there is no styling to this page

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1 +1,13 @@
...
+    <h3> <%=@post.title%> by <%= @user.name%> </h3>
+    <p> <%= @post.text%> </p>
+    <p> Comments: <%= @post.comments_counter %>, Likes: <%= @post.likes_counter %> </p>
+  </div>
+  <div>
+    <% @post.comments.each do |comment| %>
+      <p> <%= comment.user.name%>: <%= comment.text%><p>
+    <% end %>  
+  </div>
+</div>
```
Comment (tresorsawasawa) [inline app/views/posts/show.html.erb]: - [ ] Kindly make sure your `app/views/users/show.html.erb` page design sticks as much as possible to this **[wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_single_post.png?token=GHSAT0AAAAAAB3P7JH57O6URFGHTBT46NU4Y4HYOFQ)** for the project's requirements. Right now there is no styling provided yet to this page

Make sure all your pages design match its expected **[wireframes](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md)**

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1 +1,11 @@
-<h2> This is cool </h2>
\ No newline at end of file
+<section >
+  <%= render'partials/user', user: @user %>
+    <% @posts.each do |post| %>
+   <a href="/users/<%= @user.id %>/posts/<%= post.id %>">
+    <ul class="post-list">
+    <%= render 'partials/post', post: post %></ul>
+    </a>
```
Comment (tresorsawasawa) [inline app/views/posts/index.html.erb]: - [ ] Kindly maake sure you use `post.recent_five_comments` method to render **5 most recent comments** in the `app/views/posts/index.html.erb` page as required for the project. I noticed that in your `app/views/partials/_post.html.erb` partial file you render all the comments for a post. Make sure you fix this to match the requirement.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @DuaneDave 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 5 above)
Comment (DuaneDave) [reply app/views/posts/index.html.erb]: @tresorsawasawa  I have read through the requirements over again and I can't find such a requirement

Comment (tresorsawasawa) [reply app/views/posts/index.html.erb]: Hi @DuaneDave 

If you check from this **[wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_all_posts.png)** and on images below, you will notice that:
Screenshot from 2022-12-01 15-10-42
Screenshot from 2022-12-01 15-14-28

**In short, on each post you should render 5 recent comments**

Comment (DuaneDave) [reply app/views/posts/index.html.erb]: @tresorsawasawa I have added that method now

**Review by EricMbouwe — APPROVED** [review]
Hi @DuaneDave, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Addressed changes ✔
- Styling ✔
- No linter errors ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**